src/api/frameworks.ts

```diff
@@ -0,0 +1,117 @@
+import { Client } from "../apiv2";
+import { frameworksOrigin } from "../api";
+
+export const API_VERSION = "v1";
+
+const client = new Client({
+  urlPrefix: frameworksOrigin,
+  auth: true,
+  apiVersion: API_VERSION,
+});
+
+export type State = "BUILDING" | "BUILD" | "DEPLOYING" | "READY" | "FAILED";
+
+interface Codebase {
+  repository?: string;
+  rootDirectory: string;
+}
+
+/** A Stack, the primary resource of Frameworks. */
+interface Stack {
+  name: string;
+  mode?: string;
+  codebase: Codebase;
+  labels: Record<string, string>;
+  createTime: string;
+  updateTime: string;
+  uri: string;
+}
+
+export type StackOutputOnlyFields = "createTime" | "updateTime" | "uri";
+
+interface Build {
+  name: string;
+  state: State;
+  error: Status;
+  image: string;
+  source: BuildSource;
+  buildLogsUri: string;
+  createTime: Date;
+  updateTime: Date;
+  sourceRef: string;
+}
+
+export type BuildOutputOnlyFields = "createTime" | "updateTime" | "sourceRef";
+
+interface BuildSource {
+  codeBaseSource?: CodebaseSource;
+}
+
+interface Status {
+  code: number;
+  message: string;
+  details: any[];
+}
+
+interface CodebaseSource {
+  // oneof reference
+  branch: string;
+  commit: string;
+  tag: string;
+  // end oneof reference
+}
+
+export interface OperationMetadata {
+  createTime: string;
+  endTime: string;
+  target: string;
+  verb: string;
+  statusDetail: string;
+  cancelRequested: boolean;
+  apiVersion: string;
+}
+
+export interface Operation {
+  name: string;
+  metadata?: OperationMetadata;
+  done: boolean;
+  // oneof result
```

Comment:
q: what do these mean 🤔 ?

Reply:
These are part of Long Running Operation object and they are returned as response from the rpc method, 
name: Service assigned name which is unique. For HttpMapping, the `name` is a resource name ending with `operations/{unique_id}`.
OperationMetadata: Service specific metadata associated with the operation. The is the metadata_type of the Operation.
done: If `false` then Operation is still running, if `true` operation is completed.